### DISCUSSION TITLE
Prefetch gallery images for speed and UX

### DISCUSSION THREAD

**Discussion by zaidhaan** [civitai/civitai#333] [Ideas]
Currently, gallery images (specifically, those on the `/gallery/` route, those under the `/models/` reviews work smoothly) are loaded one at a time, this can be quite slow:
![image](https://user-images.githubusercontent.com/5954994/222903986-729c51c3-42f6-4bec-bab3-1a4e66a16a8e.png)

All new images take around a second to load up (I understand this is dependent on the CDN/location). Going back to a previous one would be fine as the browser's cache would already have that (refer to the last items in the screenshot).

But even so, with the image in cache it still takes a bit for the next image to be visible (I'd guess at least 300ms), this isn't really a big issue, it's just a slight annoyance with regards to UX. Of course, if the intention is not for users to use galleries to quickly browse through images (like what's already possible in the model reviews galleries) then there would be no need for this.

**Comment by JohnMadden123456** [comment]
Prefetch worked earlier. They did change that recently, either intentionally or not.

UPD: only read your title and then answered text below, but that's almost the same as you said already

Oh and btw, I have 100Mbit internet (download speed is 8-10MB) and for most images it takes seconds to load, not milliseconds. Many of them these days are hi-res and weight 1-2MB and there's initial connection time for each of them, which is without preload is triggered each time, hence 1-3 second waiting between changing slides.
